env0/utils.go

```diff
@@ -170,48 +179,20 @@ func writeResourceData(i interface{}, d *schema.ResourceData) error {
 
 		switch fieldType.Kind() {
 		case reflect.String:
-			if err := d.Set(fieldNameSC, field.String()); err != nil {
+			if err := d.Set(fieldName, field.String()); err != nil {
 				return err
 			}
 		case reflect.Int:
-			if err := d.Set(fieldNameSC, field.Int()); err != nil {
+			if err := d.Set(fieldName, field.Int()); err != nil {
 				return err
 			}
 		case reflect.Bool:
-			if err := d.Set(fieldNameSC, field.Bool()); err != nil {
+			if err := d.Set(fieldName, field.Bool()); err != nil {
 				return err
 			}
 		case reflect.Slice:
-			switch field.Type() {
-			case reflect.TypeOf([]client.ModuleSshKey{}):
-				var rawSshKeys []map[string]string
-				for i := 0; i < field.Len(); i++ {
-					sshKey := field.Index(i).Interface().(client.ModuleSshKey)
-					rawSshKeys = append(rawSshKeys, map[string]string{"id": sshKey.Id, "name": sshKey.Name})
-				}
-				if err := d.Set(fieldNameSC, rawSshKeys); err != nil {
-					return err
-				}
-			case reflect.TypeOf([]client.Agent{}):
-				var agents []map[string]string
-				for i := 0; i < field.Len(); i++ {
-					agent := field.Index(i).Interface().(client.Agent)
-					agents = append(agents, map[string]string{"agent_key": agent.AgentKey})
-				}
-				if err := d.Set(fieldNameSC, agents); err != nil {
-					return err
-				}
-			case reflect.TypeOf([]string{}):
-				var strs []interface{}
-				for i := 0; i < field.Len(); i++ {
-					str := field.Index(i).Interface().(string)
-					strs = append(strs, str)
-				}
-				if err := d.Set(fieldNameSC, strs); err != nil {
-					return err
-				}
-			default:
-				return fmt.Errorf("internal error - unhandled slice type %v", field.Type())
+			if err := writeResourceDataSlice(field.Interface(), fieldName, d); err != nil {
```

Comment:
This new method should cover all the above cases. There should be unit tests for all these cases.